keps/sig-node/3695-pod-resources-for-dra/README.md

```diff
@@ -0,0 +1,374 @@
+
+## Motivation
+
+One of the primary motivations for this KEP is to extend the PodResources API to allow node monitoring agents to access information about allocated resources by DRA. The PodResources API is also being used by CNI meta-plugins like [multus](https://github.com/k8snetworkplumbingwg/multus-cni) and [DANM](https://github.com/nokia/danm) to add resources allocated by device plugins as CNI arguments. This extension can also be used to allow these CNI plugins to reference resources allocated by DRA as well. Additional extensions to API will also make it easier for CNI meta-plugins to access resources allocated to a specific pod, rather than having to filter through resources for all pods on the node.
+
+### Goals
+
+- To allow node monitoring agents to know the alloacted DRA resources for Pods on a node.
+- Enabling the DRA feature to work with CNIs that require complex network devices such as RDMA. DRA resource drivers will allocate the resources, and the meta-plugin will read the allocated [CDI Device](https://github.com/container-orchestrated-devices/container-device-interface) using the Pod resource API. The meta-plugin will then inject the device-id as CNI arguments and invoke other CNIs (just as it does for devices allocated by the device plugin today).
+
+### Non-Goals
+
+To enhance the GetAllocatableResources() call in the PodResources API to account for resources managed by DRA. With DRA there is no standard way to get the capacity, for example.
+
+## Proposal
+
+### Risks and Mitigations
+
+This API is read-only, which removes a large class of risks. The aspects that we consider below are as follows:
+- What are the risks associated with the API service itself?
+- What are the risks associated with the data itself?
+
+| Risk                                                      | Impact        | Mitigation |
+| --------------------------------------------------------- | ------------- | ---------- |
+| Too many requests risk impacting the kubelet performances | High          | Implement rate limiting and or passive caching, follow best practices for gRPC resource management. |
+| Improper access to the data | Low | Server is listening on a root owned unix socket. This can be limited with proper pod security policies. |
+
+## Design Details
+
+### Proposed API
+
+
+Our proposal is to extend the existing pod resources gRPC service of the Kubelet by adding the CDIDevices field to the ContainerResources message. This new field will contain information about the DRA driver name, the DRA resource class, and a list of CDI Devices allocated by that DRA driver. It's important to note that this proposal focuses on retrieving CDI devices allocated by DRA. Other mechanisms that may utilize CDI, such as device plugin, are not included in the scope of this proposal.
```

Comment:
Please update this text, now that `DriverName` has been removed from the struct and `ClaimName` has been added.